python/lsst/daf/butler/registry/databases/postgresql.py

```diff
@@ -197,7 +200,7 @@ def _convertExclusionConstraintSpec(
         metadata: sqlalchemy.MetaData,
     ) -> sqlalchemy.schema.Constraint:
         # Docstring inherited.
-        args = []
+        args: list[tuple[sqlalchemy.schema.Column, str]] = []
```

Comment:
Are you surprised this method never gets called?

Reply:
I'm surprised.  I'd expect this to be called whenever we register a calibration dataset type with a new set of dimensions.

Reply:
Grepping through the code it looks like that it is only called when `TableSpec.exclusion` is not empty, and that could only happen when timespan representation class `hasExclusionConstraint()` returns `True`. I found only one implementation of `hasExclusionConstraint()` which always returns `False`.

Reply:
Oh, interesting.  The PostgreSQL-specific timespan representation should have `hasExclusionConstraint()` return `True`, as that's one of the things that `btree_gist` brings, and would avoid a table lock (IIRC) in `certify` (which I thought we were already avoiding for PostgreSQL).

But I'm not sure if we can just fix that without a migration now that we've made the mistake.